components/Header/index.js

```diff
@@ -0,0 +1,135 @@
+import { StyledCloseButton } from "../NewGoalDetails/NewGoalDetails";
+import { EmptyStateMessage } from "@/pages/statistics";
+import { Input } from "../NewGoalForm";
+import { NavBar } from "../Navigation";
+import Button from "@/components/Button";
+import Image from "next/image";
+import { useState, useEffect } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import { StyledModalBody, StyledModal, StyledOverlay } from "../Modal";
+import styled from "styled-components";
+
+const GreetingBox = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 10px;
+`;
+
+const Greeting = styled.p`
+  margin: 0;
+`;
+
+const ProfileImage = styled.div`
+  background-color: #e1d0e9;
+  padding: 8px 6px 2px;
+  border-radius: 50%;
+`;
+
+export default function Header() {
+  const [isHeaderModalOpen, setHeaderModalOpen] = useState(false);
+  const [isHeaderEditing, setHeaderEditing] = useState(false);
+  const [greeting, setGreeting] = useLocalStorageState("user greeting", {
+    defaultValue: "User",
+  });
+
+  function handleOpenHeaderModal() {
+    setHeaderModalOpen(true);
+  }
+
+  function handleCloseHeaderModal(event) {
+    event.stopPropagation();
+    setHeaderModalOpen(false);
+    setHeaderEditing(false);
+  }
+
+  useEffect(() => {
+    const handleEscapeKeyPress = (e) => {
+      if (e.key === "Escape" && isHeaderModalOpen) {
+        handleCloseHeaderModal();
+      }
+    };
+
+    document.addEventListener("keydown", handleEscapeKeyPress);
+
+    return () => {
+      document.removeEventListener("keydown", handleEscapeKeyPress);
+    };
+  });
```

Comment:
You may want to add an empty dependency array here since you want to attach the event listener only once.